docs/paper/dev/api/command-api/basics/command-executors.mdx

```diff
@@ -0,0 +1,161 @@
+
+## Example: Flyspeed command
+In the [Arguments and Literals](./arguments-and-literals) chapter, we have briefly declared the structure for a `/flyspeed` command with the use of a ranged float argument.
+But that command does not actually set the flyspeed of the executing player. In order to do that, we'd have to append an executor onto it, like this:
+
+```java title="FlightSpeedCommand.java"
+Commands.literal("flyspeed")
+    .then(Commands.argument("speed", FloatArgumentType.floatArg(0, 1.0f))
+        .executes(ctx -> {
+            float speed = ctx.getArgument("speed", float.class); // Retrieve the speed argument
+            // highlight-next-line
+            CommandSender sender = ctx.getSource().getSender(); // Retrieve the command sender
+            // highlight-next-line
+            Entity executor = ctx.getSource().getExecutor(); // Retrieve the command executor, which may or may not be the same as the sender
+
+            // Check whether the executor is a player, as you can only set a player's flight speed
+            if (executor instanceof Player player) {
+                player.setFlySpeed(speed);
+                
+                if (sender == executor) {
+                    // If the player executed the command themselves
+                    player.sendMessage("Successfully set your flight speed to " + speed);
+                }
+                else {
+                    // If the speed was set by a different sender (Aka via /execute)
+                    sender.sendMessage("Successfully set " + player.getName() + "'s flight speed to " + speed);
+                    player.sendMessage("Your flight speed has been set to " + speed);
+                }
+            }
+            else {
+                // If a non-player tried to set their own flight speed
+                sender.sendMessage("Consoles can't fly!");
+            }
+
+            return Command.SINGLE_SUCCESS;
+        })
+    );
+```
+
+### Explanation
+There is a lot to unpack, so let's break it down, top to bottom:
+
+The first lines define a `/flyspeed` command root, with a float argument named "speed", which only allows values between 0 and 1.
+We then add an executes clause to our argument branch and retrieves the speed argument. We put in `float.class` so that the return value (from the `V getArgument(String, Class<V>)` 
+method) is the correct type.
+
+Note the highlighted lines. We first retrieve the `CommandSourceStack` from our `CommandContext<CommandSourceStack>` and then finally retrieve its sender and executor.
+A `CommandSender` is an interface, which declares the `sendMessage(...)`, `getServer()`, and `getName()` methods. It is implemented by all entities, including players, 
+and the ConsoleCommandSender, which is used if a console executes a command.
+
+Next up we check whether our executor object is also instance of a Player interface. If executor were null, this would be false, which is why we require no null check.

```

Comment:
```suggestion
Next up we check whether our executor object is also instance of a `Player` interface. If executor were null, this would be false, which is why we require no null check.
```